src/mscorlib/src/System/Collections/Generic/Dictionary.cs

```diff
@@ -243,6 +244,10 @@ private struct Entry {
         }
 
         public bool ContainsValue(TValue value) {
```

Comment:
I don't think that optimizing ContainsValue it's something that's worth doing. It's not commonly used (or at least it shouldn't be commonly used) and its performance is poor anyway since it's a linear search.